soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -46,76 +167,634 @@ Color_RGBA8 Color_Skipped_Main_Default              = { 160, 160, 160, 255 };
+                gSaveContext.checkTrackerData[i].status = saveContext->checkTrackerData[i].status = RCSHOW_SAVED;
+                UpdateAllOrdering();
+                UpdateInventoryChecks();
+            } else {
+                saveContext->checkTrackerData[i].status = RCSHOW_SCUMMED;
+            }
+        }
+        SaveManager::Instance->SaveStruct("", [&]() {
+            SaveManager::Instance->SaveData("status", saveContext->checkTrackerData[i].status);
+            SaveManager::Instance->SaveData("skipped", saveContext->checkTrackerData[i].skipped);
+            SaveManager::Instance->SaveData("price", saveContext->checkTrackerData[i].price);
+            SaveManager::Instance->SaveData("hintItem", saveContext->checkTrackerData[i].hintItem);
+        });
+    });
+}
+
+void SaveFile(SaveContext* saveContext, int sectionID, bool fullSave) {
+    SaveTrackerData(saveContext, sectionID, fullSave);
+    if (fullSave) {
+        savedFrames = 40;
+    }
+}
+
+void LoadFile() {
+    Teardown();
+    LoadSettings();
+    TrySetAreas();
+    SaveManager::Instance->LoadArray("checks", RC_MAX, [](size_t i) {
+        SaveManager::Instance->LoadStruct("", [&]() {
+            SaveManager::Instance->LoadData("status", gSaveContext.checkTrackerData[i].status);
+            SaveManager::Instance->LoadData("skipped", gSaveContext.checkTrackerData[i].skipped);
+            SaveManager::Instance->LoadData("price", gSaveContext.checkTrackerData[i].price);
+            SaveManager::Instance->LoadData("hintItem", gSaveContext.checkTrackerData[i].hintItem);
+        });
+        RandomizerCheckTrackerData entry = gSaveContext.checkTrackerData[i];
+        RandomizerCheck rc = static_cast<RandomizerCheck>(i);
+        if (rc == RC_UNKNOWN_CHECK || rc == RC_MAX ||
+            !RandomizerCheckObjects::GetAllRCObjects().contains(rc))
+            return;
+
+        RandomizerCheckObject entry2;
+        if (rc == RC_GIFT_FROM_SAGES && !gSaveContext.n64ddFlag) {
+            entry2 = RCO_RAORU;
+        } else {
+            entry2 = RandomizerCheckObjects::GetAllRCObjects().find(rc)->second;
+        }
+        if (!IsVisibleInCheckTracker(entry2)) return;
+
+        checksByArea.find(entry2.rcArea)->second.push_back(entry2);
+        if (entry.status == RCSHOW_SAVED || entry.skipped) {
+            areaChecksGotten[entry2.rcArea]++;
+        }
+
+        if (areaChecksGotten[entry2.rcArea] != 0 || RandomizerCheckObjects::AreaIsOverworld(entry2.rcArea)) {
+            areasSpoiled |= (1 << entry2.rcArea);
+        }
+    });
+    if (OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_LINKS_POCKET) != RO_LINKS_POCKET_NOTHING && gSaveContext.n64ddFlag) {
+        s8 startingAge = OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_STARTING_AGE);
+        RandomizerCheckArea startingArea;
+        switch (startingAge) {
+            case RO_AGE_CHILD:
+                startingArea = RCAREA_KOKIRI_FOREST;
```

Comment:
I think it would be better if I could figure out the area you're actually starting in for this, tbh, than having it in an agnostic place. Gift from Sages is just in Temple of Time, since it's closest to where you actually get it.

Reply:
Currently our Gift from Sages is tied to LACS, but after the planned rando rework, I'd like to decouple it from LACS like it is on N64 (granted immediately regardless of where you are once conditions are met), which would make it no longer tied to temple of time. I figured that location agnostic handling would be nice.

This is not a blocker to me, and if we would rather handle that once it happens, I'm fine,